src/extension/editor.ts

```diff
@@ -248,6 +243,83 @@ export class Zone extends ZoneBase {
     }
 }
 
+export class SelectionZone extends Zone {
+    private animationRunning: boolean = false;
+
+    constructor(layoutItem: LayoutItem, parent: ZoneGroup | null, stage: ClutterActor) {
+        super(layoutItem, parent, stage);
+        const color = this.widget.get_theme_node().get_background_color();
+        let newAlpha = Math.min(color.alpha + 35, 255);
+        // since an alpha value lower than 160 is not so much visible, enforce a minimum value of 160
+        if (newAlpha < 160) newAlpha = 160;
+        // The final alpha value is divided by 255 since CSS needs a value from 0 to 1, but ClutterColor expresses alpha from 0 to 255
+        this.widget.set_style(`
+            background-color: rgba(${color.red}, ${color.green}, ${color.blue}, ${newAlpha / 255}) !important;
+        `);
+    }
+
+    public move(newX: number, newY: number, newWidth: number, newHeight: number, ease: boolean = true) {
+        if (this.animationRunning) return; // if the animation is still running, do not animate again
+
+        // if both position and widths are not changed
+        // then there is nothing to move o  r resize
+        if (newX == this.x && 
+            newY == this.y &&
+            newWidth == this.width &&
+            newHeight == this.height) {
+            return;
+        }
+        
+        // if the zone was never shown before, start the animation from cursor's coordinates
+        if (this.innerWidth <= 0) {
+            let [x, y] = global.get_pointer();
+            this.widget.x = x;
+            this.widget.y = y;
+            this.widget.width = 0;
+            this.widget.height = 0;
+        }
+
+        // update location
+        this.x = newX;
+        this.y = newY;
+        // update sizes
+        this.width = newWidth;
+        this.height = newHeight;
+
+        this.widget.visible = true;
+        if (ease) { // animate with easing
+            this.animationRunning = true;
+            (this.widget as any).ease({
+                time: ANIMATION_SPEED,
```

Comment:
Shouldn't it be `duration: ANIMATION_SPEED` ?
I'm writing this because while I was testing, I thought that the animation felt "too slow" for my personal preference. It feels a lot faster by setting the duration.

Reply:
You are right, thank you! It should be `duration: ANIMATION_SPEED`!